game/src/edit/new_traffic_signals.rs

```diff
@@ -0,0 +1,246 @@
+        let phase_btn = Btn::custom(unselected_btn, selected_btn, bbox).build(
+            ctx,
+            format!("phase {}", idx + 1),
+            None,
+        );
+
+        let phase_col = Widget::col(vec![
+            Widget::row(vec![
+                // TODO Print duration
+                Line(format!("Phase {}", idx + 1)).small_heading().draw(ctx),
+                Btn::svg_def("system/assets/tools/edit.svg").build(
+                    ctx,
+                    format!("change duration of phase {}", idx + 1),
+                    if selected == idx {
+                        hotkey(Key::X)
+                    } else {
+                        None
+                    },
+                ),
+                if num_phases > 1 {
+                    Btn::svg_def("system/assets/tools/delete.svg")
+                        .build(ctx, format!("delete phase {}", idx + 1), None)
+                        .align_right()
+                } else {
+                    Widget::nothing()
+                },
+            ]),
+            Widget::row(vec![
+                phase_btn,
+                Widget::col(vec![
+                    if idx == 0 {
+                        Btn::text_fg("↑").inactive(ctx)
+                    } else {
+                        Btn::text_fg("↑").build(ctx, format!("move up phase {}", idx + 1), None)
+                    },
+                    if idx == num_phases - 1 {
+                        Btn::text_fg("↓").inactive(ctx)
+                    } else {
+                        Btn::text_fg("↓").build(ctx, format!("move down phase {}", idx + 1), None)
+                    },
+                ])
+                .centered_vert()
+                .align_right(),
+            ]),
+        ])
+        .padding(10);
+
+        if idx == selected {
+            col.push(phase_col.bg(Color::hex("#2A2A2A")));
+        } else {
+            col.push(phase_col);
+        }
+    }
+
+    Composite::new(Widget::col(col))
+        .aligned(HorizontalAlignment::Left, VerticalAlignment::Top)
+        .exact_size_percent(30, 85)
+        .build(ctx)
+}
+
+fn draw_multiple_signals(
+    ctx: &mut EventCtx,
+    app: &App,
+    members: &BTreeSet<IntersectionID>,
+    idx: usize,
+) -> GeomBatch {
+    let mut batch = GeomBatch::new();
+    for i in members {
+        batch.push(
+            app.cs.normal_intersection,
+            app.primary.map.get_i(*i).polygon.clone(),
+        );
+
+        draw_signal_phase(
+            ctx.prerender,
+            app.primary
+                .map
+                .get_traffic_signal(*i)
+                .phases
+                .get(idx)
+                .unwrap_or(&Phase::new()),
+            *i,
+            None,
+            &mut batch,
+            app,
+            TrafficSignalStyle::Sidewalks,
+        );
+    }
+
+    // Transform to a screen-space icon. How much should we scale things down?
+    batch = batch.autocrop();
+    let mut zoom: f64 = 1.0;
+    if true {
```

Comment:
Both of these aren't quite right. In the downtown example, most of the space is empty between the 3 intersections. Maybe there's a way to reduce the empty space, while preserving the relative orientation of the 3 groups of polygons.